app/controllers/transcriptions_controller.rb

```diff
@@ -41,6 +52,15 @@ def update
     render jsonapi: @transcription
   end
 
+  def unlock
+    @transcription = Transcription.find(params[:id])
+    authorize @transcription, :update?
+
+    return unless @transcription.locked_by == current_user.login
```

Comment:
what does a return look like here at the serializer level?

Reply:
Nothing is serialized if the method exits here. The expectation is that the front-end will send a patch request to the unlock endpoint if a user with edit permissions navigates away from the transcription edit page. That line exists for the case where editor-user-1 has the transcription locked, and editor-user-2 navigates to the page, sees the "transcription is locked" message, then navigates away. This would trigger a patch request to unlock, but we dont want to allow anyone other than the locking user to unlock the transcription.

Perhaps it would be best to throw an error here rather than just return/exit?

I'm also thinking it would be best to give the front-end instructions to only make a request to the unlock method if the transcription isn't locked by a different user.

Reply:
>Perhaps it would be best to throw an error here rather than just return/exit?

Yes - try to be informative to the client as to why there request failed, use HTTP status codes and non-cryptic error messages. 

Do this and the folks interacting with the API will love you, they will speak about you with reverence and at night, they will tell tall tales about your feats of coding...

Reply:
>Perhaps it would be best to throw an error here rather than just return/exit?

Yes - try to be informative to the client as to why there request failed, use HTTP status codes and non-cryptic error messages. 

Do this and the folks interacting with the API will love you, they will speak about you with reverence and at night, they will tell tall tales about your feats of coding...